### PR TITLE
Warn when using “Require” in a section

### LINKED ISSUE #6594: Warnings generated by the standard library
#### Version

8.8+alpha

#### Description of the problem

The compilation generates the following warnings:

```
Warning: Summary was captured out of module scope for entry field-tac-carrier-table
Warning: Summary was captured out of module scope for entry field-tac-name-table
Warning: Summary was captured out of module scope for entry ring-tac-carrier-table
Warning: Summary was captured out of module scope for entry ring-tac-name-table
```

According to @ejgallego, it should be made into a debug warning and disabled by default.

### PR DESCRIPTION
**Pull request by vbgl**
Fixes #6594.

- [ ] Corresponding documentation was added / updated.
- [x] Entry added in CHANGES.

### CODE REVIEW

File: library/summary.ml [hunk 1]
```diff
@@ -89,6 +89,13 @@ let unfreeze_single name state =
       Pp.(seq [str "Error unfreezing summary "; str name; fnl (); CErrors.iprint e]);
     iraise e
 
+let warn_summary_out_of_scope =
+  let name = "summary-out-of-scope" in
+  let category = "pedantic" in
+  let default = CWarnings.Disabled in
+  CWarnings.create ~name ~category ~default (fun name ->
+       Pp.(str "Summary was captured out of module scope for entry " ++ str name))
```
Comment (ejgallego) [inline library/summary.ml]: I'd suggest a more informative message, such as "A Coq plugin was loaded inside a local scope (such as a Section). It is recommended to load plugins at the start of the file.

Comment (maximedenes) [reply library/summary.ml]: I was expecting that too, but there is another warning below. Maybe this one is capturing other scenarios?

Comment (ejgallego) [reply library/summary.ml]: Well for this code I would indeed do just a `msg_debug` flag, as I see it more as a dev-only warning. Once we get the configure profiles in place it should be easy to tweak this stuff better.

Comment (maximedenes) [reply library/summary.ml]: What I suggest is to keep using a warning, but have a dedicated "dev" category for them. Those warnings should be off by default, and turned on for the developer profile.

Comment (maximedenes) [reply library/summary.ml]: > "dev"

(feel free to pick another name)

**Comment by vbgl** [comment]
I just rephrased the warning about the summary and moved it to the brand new “dev” category.

**Comment by Zimmi48** [comment]
@vbgl Could you squash b700a4b and 985c980 together?

**Comment by vbgl** [comment]
Squashed.

**Comment by Zimmi48** [comment]
Thanks! Feel free to remove the label yourself in these cases.

**Comment by maximedenes** [comment]
https://travis-ci.org/vbgl/coq/builds/350432928 looks good.

**Pull request merged**